packages/editor/src/components/page-attributes/parent.js

```diff
@@ -1,92 +1,148 @@
 /**
  * External dependencies
  */
-import { get } from 'lodash';
+import {
+	get,
+	unescape as unescapeString,
+	debounce,
+	flatMap,
+	repeat,
+} from 'lodash';
 
 /**
  * WordPress dependencies
  */
 import { __ } from '@wordpress/i18n';
-import { TreeSelect } from '@wordpress/components';
-import { compose } from '@wordpress/compose';
-import { withSelect, withDispatch } from '@wordpress/data';
+import { ComboboxControl } from '@wordpress/components';
+import { useState, useMemo } from '@wordpress/element';
+import { useSelect, useDispatch } from '@wordpress/data';
 
 /**
  * Internal dependencies
  */
 import { buildTermsTree } from '../../utils/terms';
 
-export function PageAttributesParent( {
-	parent,
-	postType,
-	items,
-	onUpdateParent,
-} ) {
+function getTitle( post ) {
+	return post?.title?.rendered
+		? post.title.rendered
+		: `#${ post.id } (${ __( 'no title' ) })`;
+}
+
+export function PageAttributesParent() {
+	const { editPost } = useDispatch( 'core/editor' );
+	const [ fieldValue, setFieldValue ] = useState( false );
+	const {
+		parentPost,
+		parentPostId,
+		isLoading,
+		parent,
+		items,
+		postType,
+	} = useSelect(
+		( select ) => {
+			const { getPostType, getEntityRecords, getEntityRecord } = select(
+				'core'
+			);
+			const { isResolving } = select( 'core/data' );
+			const { getCurrentPostId, getEditedPostAttribute } = select(
+				'core/editor'
+			);
+			const postTypeSlug = getEditedPostAttribute( 'type' );
+			const pageId = getEditedPostAttribute( 'parent' );
+			const pType = getPostType( postTypeSlug );
+			const postId = getCurrentPostId();
+			const isHierarchical = get( pType, [ 'hierarchical' ], false );
+			const query = {
+				per_page: 100,
+				exclude: postId,
+				parent_exclude: postId,
+				orderby: 'menu_order',
+				order: 'asc',
+				_fields: 'id,title,parent',
+			};
+			if ( parentPost && fieldValue && '' !== fieldValue ) {
+				query.search = fieldValue;
+			}
+			const theParentID = getEditedPostAttribute( 'parent' );
```

Comment:
This is still duplicate. `theParentID` and `pageId` are the same and returned under two fields `parentPostId` and `parent` respectively.